crates/libs/bindgen/src/extensions/in6_addr.rs

```diff
@@ -0,0 +1,28 @@
+use super::*;
+
+pub fn gen() -> TokenStream {
+    quote! {
+        impl ::core::convert::From<::std::net::Ipv6Addr> for IN6_ADDR {
+            fn from(addr: ::std::net::Ipv6Addr) -> Self {
+                Self { u: IN6_ADDR_0 { Byte: addr.octets() } }
+            }
+        }
+        impl ::core::convert::From<&::std::net::Ipv6Addr> for IN6_ADDR {
```

Comment:
Here the same comment applies: https://github.com/microsoft/windows-rs/pull/1756#discussion_r871508252. `Ipv4Addr`/`Ipv6Addr`/`IpAddr`/`SocketAddrV4`/`SocketAddrV6` are `Copy`. Also, `IN_ADDR`/`IN6_ADDR`/`SOCKADDR_IN`/`SOCKADDR_IN6`/`SOCKADDR_INET` are `Copy`, too.

So there is no strict necessity to implement `From` for both a reference and a value for these types. Implementing `From` for values would be enough.
For example, in std [`impl From<Ipv4Addr> for u32`](https://doc.rust-lang.org/std/primitive.u32.html#impl-From%3CIpv4Addr%3E) is implemented only for a value, but not for a reference. I think, that the same logic can be applied here.

Reply:
I'd noticed that for `u32` as well and I'm glad there's consensus on the topic.

Reply:
`SocketAddrVN.addr()` returns a reference to an `IpvNAddr`, so I had to deref it the ugly way :-D